index.js

```diff
@@ -223,6 +225,7 @@ async function run() {
     );
   } catch (error) {
     core.setFailed(`Failed to create server: ${error.message}`);
+    throw error;
```

Comment:
I think this is the highest-level `try`/`catch` block, so we shouldn't re-throw here.  This is where we want to mark the action failed and let the process exit on its own, and I believe we've seen that work as expected when the top-level timeout is hit.